src/status_im/multiaccounts/core.cljs

```diff
@@ -43,13 +43,18 @@
 ;; available
 (defn contact-two-names
   "Returns vector of two names in next order nickname, ens name, display-name, three word name, public key"
-  [{:keys [names public-key] :as contact} public-key?]
+  [{:keys [names
+           compressed-key
+           public-key]
+    :as   contact} public-key?]
   (let [{:keys [nickname
                 ens-name
                 display-name
                 three-words-name]}
         (or names (contact-names contact))
-        short-public-key (when public-key? (utils/get-shortened-address public-key))]
+        short-public-key (when public-key?
+                           (utils/get-shortened-address (or compressed-key
+                                                            public-key)))]
     (->> [nickname ens-name display-name three-words-name short-public-key]
          (remove string/blank?)
          (take 2))))
```

Comment:
I can see that this `contact-two-names` function was already accepting a `public-key?` parameter, so I wouldn't suggest refactoring this in this PR, but it does seem like `public-key?` is unnecessary since the fn could see if a public-key is passed in and decide accordingly what to return.

Nice use of FP in threading macro (not new, i know)